ompi/mca/coll/base/coll_base_topo.c

```diff
@@ -59,7 +59,12 @@ static int calculate_num_nodes_up_to_level( int fanout, int level )
 {
     /* just use geometric progression formula for sum:
        a^0+a^1+...a^(n-1) = (a^n-1)/(a-1) */
-    return ((pown(fanout,level) - 1)/(fanout - 1));
+    if (1 == fanout) {
+        return 1;
+    }
```

Comment:
Upon further reivew, from the conversation, I'm unsure if this should `return 1;` or `return level;` It sounds like it should `return level;` since that's the number of nodes preceding this level when the fanout is 1. Is that correct?

Reply:
yes, `level` should be the outcome here.